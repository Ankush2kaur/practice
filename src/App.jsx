
import './App.css'
import StudentForm from './Components/StudentForm';
import StudentList from './Components/StudentList'
import 'bootstrap/dist/css/bootstrap.min.css';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom'
import Update from './Components/Update';
import Read from './Components/Read';


function App() {

  return (
    <>
  <Router>
      <Routes>
        <Route path="/" element={<StudentList/>} />
        <Route path="/add-student" element={<StudentForm/>} />
        <Route path="/update/:id" element={<Update/>}/>
        <Route path="/read/:id" element={<Read/>}/>
      </Routes>
    </Router>
    </>
  )
}

export default App
