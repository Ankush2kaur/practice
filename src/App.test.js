import {fireEvent, render,screen} from "@testing-library/react";
import App from "./App";
import { MemoryRouter, Routes, Route } from 'react-router-dom';
// import FunctionalTesting from "./FunctionalTesting";
// import handleOtherMethod from "./Helper";
// // beforeAll(()=>{
//   console.log("****before al Hook***")
// })
// beforeEach(()=>{
//   console.log("****before al Hook***")
// })

// test("Testing First React App Case",()=>{
//   render(<App/>);
//   const text= screen.getByText(/First React Test case/i);
//  const title = screen.getByTitle("Cat img")
//   const text2=screen.getByText("Hello")
// expect (text).toBeInTheDocument();
// expect (text2).toBeInTheDocument();
// expect(title).toBeInTheDocument();
// })

// test("Testing Input Box",()=>{
//   render(<App/>)
// let checkInput = screen.getByRole("textbox");
// let checkInputPlaceholder = screen.getByPlaceholderText("Enter User name");
// expect(checkInputPlaceholder).toBeInTheDocument();
// expect (checkInput).toBeInTheDocument();
// expect(checkInput).toHaveAttribute("name","username")
// expect(checkInput).toHaveAttribute("id","userId")
// expect(checkInput).toHaveAttribute("type","text")
// expect(checkInput).toHaveAttribute("value","Ankush Kaur")
// });


// test("On change event Testing",()=>{
//   render(<App/>)
//   let input = screen.getByRole("textbox");
//   fireEvent.change(input,{target:{value:"abc"}});
//   expect(input.value).toBe("abc");
// })

// describe.only("oN cLICK tEST",()=>{

//   test("Onclick button test",()=>{
//   render(<App/>)
//   let button = screen.getByRole("button");
//   fireEvent.click(button);
//   expect(screen.getByText("updated DATA")).toBeInTheDocument();
//   })
// })


// test("snapshot testing",()=>{
// const view = render(<App/>)
// expect (view).toMatchSnapshot();
// })

// test("method Testing case 1",()=>{
//   render(<FunctionalTesting/>);
//   const btn= screen.getAllByTestId("btn1");
//   fireEvent.click(btn);
//   expect(screen.getByAltText("hello")).toBeInTheDocument();
// })


// test("method testing case2",()=>{
// expect(handleOtherMethod
//   ()).toMatch("hi")
// }
// )
// test ("get by role",()=>{
// render(<FunctionalTesting/>);
// const inputFiled= screen.getByRole("textbox")
// const btn = screen.getByRole("button");
// expect(inputFiled).toBeInTheDocument();
// expect(inputFiled).toHaveValue("hello");
// expect(inputFiled).toBeDisabled();
// expect (btn).toBeInTheDocument();
// })

  
// mocking of components 
jest.mock('./Components/StudentList', () => () => <div>StudentList Component</div>);

describe("testing Routing of whole App",()=>{
 test("Rendering Student Componennt on Root path",()=>{
 render(  <MemoryRouter initialEntries={['/']}>
     <App />
   </MemoryRouter>)
 })
expect(screen.getByText("StudentList Component")).toBeInTheDocument();
}
)
