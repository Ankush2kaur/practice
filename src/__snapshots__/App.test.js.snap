// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <p>
          First React Test Case
        </p>
        <p>
          Hello
        </p>
        <input
          id="userId"
          name="username"
          placeholder="Enter User name"
          type="text"
          value=""
        />
        <br />
        <br />
        <img
          alt="nature"
          src="https://images.unsplash.com/photo-1518791841217-8f162f1e1131"
          style="height: 200px; width: 400px;"
          title="Cat img"
        />
        <button>
          Update Data
        </button>
        <h1 />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <p>
        First React Test Case
      </p>
      <p>
        Hello
      </p>
      <input
        id="userId"
        name="username"
        placeholder="Enter User name"
        type="text"
        value=""
      />
      <br />
      <br />
      <img
        alt="nature"
        src="https://images.unsplash.com/photo-1518791841217-8f162f1e1131"
        style="height: 200px; width: 400px;"
        title="Cat img"
      />
      <button>
        Update Data
      </button>
      <h1 />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
